### PR TITLE
RD-173 - Stage should be able to work without db proxy

### PR DESCRIPTION
**Pull request by qooban**
### Changes
- Updated pg and sequelize to latest versions
- Updated DB connection setup algorithm (see [RD-173](https://cloudifysource.atlassian.net/browse/RD-173) for requirements)
- Adapted migration script to refactored DB connection setup

### CODE REVIEW

File: backend/db/Connection.js [hunk 1]
```diff
@@ -1,51 +1,160 @@
...
-    {
-        dialect: 'postgres',
-        pool: {
-            max: 5,
-            min: 0,
-            idle: 10000
+function getDbOptions(configOptions) {
+    const options = _.merge(
+        {
+            logging: message => logger.debug(message)
```
Comment (kubama) [inline backend/db/Connection.js:15]:
```suggestion
            logging: logger.debug
```

File: backend/db/Connection.js [hunk 2]
```diff
@@ -1,51 +1,160 @@
...
+                logger.info(`Checking DB host ${i}: ${dbHost}`);
+                const hasResponded = await isResponding(dbHost);
+                if (hasResponded) {
+                    logger.debug(`DB host ${dbHost} has responded.`);
+                    respondingHost = urls[i];
+                    break;
+                } else {
+                    logger.debug(`DB host ${dbHost} not responding.`);
+                }
+            }
```
Comment (kubama) [inline backend/db/Connection.js:73]: I'd use `_.find` here

File: backend/db/Connection.js [hunk 3]
```diff
@@ -1,51 +1,160 @@
...
+    let selectedDbUrl = null;
+
+    if (_.isString(dbUrls)) {
+        selectedDbUrl = dbUrls;
+    } else if (_.isArray(dbUrls)) {
+        logger.info('Selecting DB host...');
+        selectedDbUrl = await findRespondingHost(dbUrls);
+    } else {
+        throw new Error(
+            'Invalid type of db.url parameter passed to the configuration. Expected string or list of strings.'
```
Comment (kubama) [inline backend/db/Connection.js:94]:
```suggestion
            'Invalid type of db.url parameter passed to the configuration. Expected string or array of strings.'
```

File: backend/server.js [hunk 4]
```diff
@@ -2,26 +2,25 @@
...
-const ToursHandler = require('./handler/ToursHandler');
-const WidgetHandler = require('./handler/WidgetHandler');
-const TemplateHandler = require('./handler/TemplateHandler');
-const app = require('./app');
-
-module.exports = Promise.all([ToursHandler.init(), WidgetHandler.init(), TemplateHandler.init()])
+module.exports = DBConnection.init()
+    .then(() => {
+        logger.info('DB connection initialized successfully.');
+        Promise.all([ToursHandler.init(), WidgetHandler.init(), TemplateHandler.init()]);
```
Comment (kubama) [inline backend/server.js:22]: Did you mean:
```suggestion
        return Promise.all([ToursHandler.init(), WidgetHandler.init(), TemplateHandler.init()]);
```
?

File: backend/db/Connection.js (hunk 1 above)
Comment (qooban) [reply backend/db/Connection.js:15]: Adding to suggestions batch.

File: backend/server.js (hunk 4 above)
Comment (qooban) [reply backend/server.js:22]: Right. Adding to suggestions batch.

File: backend/db/Connection.js (hunk 3 above)
Comment (qooban) [reply backend/db/Connection.js:94]: Adding to suggestions batch.

File: backend/db/Connection.js (hunk 2 above)
Comment (qooban) [reply backend/db/Connection.js:73]: Could you provide working solution for `_.find` with async predicate function (without waiting for all results)?

Comment (kubama) [reply backend/db/Connection.js:73]: Please ignore my original comment :-)

File: backend/migration.js [hunk 5]
```diff
@@ -57,6 +57,10 @@ function endMigration(exitCode = 0) {
     process.exit(exitCode);
 }
 
+function getCurrent(executed) {
+    return executed.length > 0 ? _.last(executed).file : '<NO_MIGRATIONS>';
```
Comment (kubama) [inline backend/migration.js:61]:
```suggestion
    return _.last(executed).file || '<NO_MIGRATIONS>';
```

File: backend/test/routes/Applications.test.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
+const request = require('supertest');
+const mockDb = require('../mocks/mockDb');
+require('../mocks/passport');
+
+describe('/applications endpoint', () => {
+    it('allows to get all data about applications', () => {
+        mockDb({
+            Application: {
+                findAll: jest.fn(() =>
```
Comment (kubama) [inline backend/test/routes/Applications.test.js:9]: I don't think it brings any value to wrap the function with `jest.fn`. Applies to other tests as well.

**Review by kubama — APPROVED** [review]

File: backend/migration.js (hunk 5 above)
Comment (qooban) [reply backend/migration.js]: Changed. Did it as well in PR for Composer.

File: backend/test/routes/Applications.test.js (hunk 6 above)
Comment (qooban) [reply backend/test/routes/Applications.test.js]: Agree. Removed wrapping.

**Review by kubama — APPROVED** [review]

**Pull request merged**